applications/matter_bridge/src/simulated_providers/simulated_humidity_sensor_data_provider.cpp

```diff
@@ -56,16 +56,23 @@ void SimulatedHumiditySensorDataProvider::TimerTimeoutCallback(k_timer *timer)
 		return;
 	}
 
-	SimulatedHumiditySensorDataProvider *provider =
-		reinterpret_cast<SimulatedHumiditySensorDataProvider *>(timer->user_data);
-
-	/* Get some random data to emulate sensor measurements. */
-	provider->mHumidity =
-		chip::Crypto::GetRandU16() % (kMaxRandomTemperature - kMinRandomTemperature) + kMinRandomTemperature;
-
-	LOG_INF("SimulatedHumiditySensorDataProvider: Updated humidity value to %d", provider->mHumidity);
-
-	DeviceLayer::PlatformMgr().ScheduleWork(NotifyAttributeChange, reinterpret_cast<intptr_t>(provider));
+	DeviceLayer::PlatformMgr().ScheduleWork(
+		[](intptr_t p) {
+			SimulatedHumiditySensorDataProvider *provider =
+				reinterpret_cast<SimulatedHumiditySensorDataProvider *>(p);
+
+			/* Get some random data to emulate sensor measurements. */
+			provider->mHumidity =
+				chip::Crypto::GetRandU16() % (kMaxRandomTemperature - kMinRandomTemperature) +
+				kMinRandomTemperature;
+
+			LOG_INF("SimulatedHumiditySensorDataProvider: Updated humidity value to %d",
+				provider->mHumidity);
+
+			DeviceLayer::PlatformMgr().ScheduleWork(NotifyAttributeChange,
+								reinterpret_cast<intptr_t>(provider));
+		},
+		reinterpret_cast<intptr_t>(timer->user_data));
 }
 
 void SimulatedHumiditySensorDataProvider::NotifyAttributeChange(intptr_t context)
```

Comment:
BTW. Is MatterBridgedDevice::NotifyAttributeChange() used anywhere?

Reply:
Hmm, I can't remember why is it there. I will remove it, because it does not seem to be useful.